src/routing/approved.rs

```diff
@@ -665,7 +668,38 @@ impl Approved {
                 self.handle_join_request(msg.src().to_node_peer(sender)?, *join_request.clone())
             }
             Variant::UserMessage(content) => {
-                self.handle_user_message(msg.src().src_location(), *msg.dst(), content.clone())
+                if msg.dst() == &DstLocation::AccumulatingNode(self.node().name()) {
+                    if let SrcAuthority::BlsShare { proof_share, .. } = msg.src() {
+                        let signed_bytes = bincode::serialize(&msg.signable_view())?;
+                        match self
+                            .message_accumulator
+                            .add(&signed_bytes, proof_share.clone())
+                        {
+                            Ok(proof) => {
+                                trace!("Successfully accumulated message: {:?}", msg);
+                                let key = msg.proof_chain_last_key()?;
```

Comment:
Should we only check the last key?

Reply:
It's enough. The proof chain itself is already validated by this point (meaning each key is validly signed by the previous key) so all that's left to do is to verify that the last key signs the message.